lib/lib.es2015.promise.d.ts

```diff
@@ -30,117 +30,117 @@ interface PromiseConstructor {
      * a resolve callback used to resolve the promise with a value or the result of another promise,
      * and a reject callback used to reject the promise with a provided reason or error.
      */
-    new <T>(executor: (resolve: (value?: T | PromiseLike<T>) => void, reject: (reason?: any) => void) => void): Promise<T>;
+    new <T>(executor: (resolve: (value?: T | PromiseLike<T> | awaited T) => void, reject: (reason?: any) => void) => void): Promise<T>;
```

Comment:
Shouldn’t this return `Promise<awaited T>`?
```suggestion
    new <T>(executor: (resolve: (value?: T | PromiseLike<T> | awaited T) => void, reject: (reason?: any) => void) => void): Promise<awaited T>;
```

Reply:
No, that would prevent it from being subclassable. The `T` in the constructor argument list must match the `T` in the return type.